src/transform/src/join_implementation.rs

```diff
@@ -555,6 +556,31 @@ impl<'a> Orderer<'a> {
                 }
             }
         }
+
+        // For each input, store each expression that is equivalent to a literal
+        let mut literal_equivalences = vec![HashSet::new(); inputs];
+        for equivalence in equivalences.iter() {
+            // see if equivalence class contains a literal
+            if equivalence.iter().any(|expr| expr.is_literal()) {
+                for expr in equivalence.iter() {
+                    let support = expr.support();
+                    let mut support_iter = support.iter();
+                    if let Some(first) = support_iter.next() {
+                        let input = input_relation[*first];
+                        if support_iter.all(|col| input_relation[*col] == input) {
```

Comment:
could this be simplified a bit by having a `bound_by` function or something that checks if a given expression is bound entirely by one of the join inputs (I think that's what this is doing?)?